From: Stefan Fritsch <sf@debian.org>
Subject: fix some issues found by cppcheck
Date: 2010-04-21 00:36:31 +0200

diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' mp3gain-1.5.1~/id3tag.c mp3gain-1.5.1/id3tag.c
--- mp3gain-1.5.1~/id3tag.c	2010-04-21 00:36:31.810590842 +0200
+++ mp3gain-1.5.1/id3tag.c	2010-04-21 00:36:31.858533970 +0200
@@ -303,6 +303,8 @@
 				break;
 			default:
 				va_end(ap);
+				free(frame->data);
+				free(frame);
 				return NULL;
 		}
 	}
diff -urNad '--exclude=CVS' '--exclude=.svn' '--exclude=.git' '--exclude=.arch' '--exclude=.hg' '--exclude=_darcs' '--exclude=.bzr' mp3gain-1.5.1~/mp3gain.c mp3gain-1.5.1/mp3gain.c
--- mp3gain-1.5.1~/mp3gain.c	2010-04-21 00:36:31.790535993 +0200
+++ mp3gain-1.5.1/mp3gain.c	2010-04-21 00:38:28.717567899 +0200
@@ -765,6 +765,7 @@
             passError(MP3GAIN_UNSPECIFED_ERROR, 3,
                 "\nCan't open ", outfilename, " for temp writing\n");
 			NowWriting = 0;
+			free(outfilename);
 			return M3G_ERR_CANT_MAKE_TMP;
 		} 
  
@@ -782,6 +783,7 @@
 	  passError( MP3GAIN_UNSPECIFED_ERROR, 3,
           "\nCan't open ", filename, " for modifying\n");
 	  NowWriting = 0;
+	  free(outfilename);
 	  return M3G_ERR_CANT_MODIFY_FILE;
   }
   else {
